kubespawner/spawner.py

```diff
@@ -144,145 +310,139 @@ def get_pod_manifest(self):
-                        'image': self.singleuser_image_spec,
-                        'resources': {
-                            'requests': {
-                                'memory': self.mem_request,
-                                'cpu': self.cpu_request,
-                            },
-                            'limits': {
-                                'memory': self.mem_limit,
-                                'cpu': self.cpu_limit
-                            }
-                        },
-                        'env': [
-                            {'name': k, 'value': v}
-                            for k, v in self.get_env().items()
-                        ],
-                        'volumeMounts': self._expand_all(self.volume_mounts) + hack_volume_mounts
-                    }
-                ],
-                'volumes': self._expand_all(self.volumes) + hack_volumes
-            }
-        }
-
-    def _get_pod_url(self, pod_name=None):
-        url = '{host}/api/{version}/namespaces/{namespace}/pods'.format(
-            host=self.kube_api_endpoint,
-            version=self.kube_api_version,
-            namespace=self.kube_namespace
+        return make_pod_spec(
+            self.pod_name,
+            self.singleuser_image_spec,
+            self.get_env(),
+            self._expand_all(self.volumes) + hack_volumes,
+            self._expand_all(self.volume_mounts) + hack_volume_mounts,
+            self.cpu_limit,
+            self.cpu_guarantee,
+            self.mem_limit,
+            self.mem_guarantee,
         )
-        if pod_name:
-            return url + '/' + pod_name
-        return url
-
-    @property
-    def session(self):
-        if hasattr(self, '_session'):
-            return self._session
-        else:
-            self._session = FuturesSession()
-            auth_header = 'Bearer %s' % self.kube_token
-            self._session.headers['Authorization'] = auth_header
-            self._session.verify = self.kube_ca_path
-            return self._session
-
-    def load_state(self, state):
-        super(KubeSpawner, self).load_state(state)
 
-    def get_state(self):
-        state = super(KubeSpawner, self).get_state()
-        state['hi'] = 'hello'
-        return state
 
     @gen.coroutine
     def get_pod_info(self, pod_name):
-        resp = self.session.get(
-            self._get_pod_url(),
-            params={'labelSelector': 'name = %s' % pod_name})
-        data = yield resp
-        return data.json()
+        """
+        Fetch info about a specific pod with the given pod name in current namespace
 
-    def is_pod_running(self, pod_info):
-        return 'items' in pod_info and len(pod_info['items']) > 0 and \
-            pod_info['items'][0]['status']['phase'] == 'Running'
+        Return `None` if pod with given name does not exist in current namespace
+        """
+        try:
+            resp = yield self.httpclient.fetch(self.request(
```

Comment:
Naming suggestion (for user friendliness):

I tend to prefer `response` to `resp`. Either is fine.